pyrobosim/pyrobosim/utils/traversability_graph.py

```diff
@@ -130,3 +130,35 @@ def nearest(self, pose):
                 min_dist = dist
                 n_nearest = n
         return n_nearest
+
+    def get_neighbors(self, node):
+        """
+        Returns a list of the neighbors for given node
+
+        :param node: The node who neighbors are required
+        :param type: :class: `pyrobosim.utils.traversability_graph.Node`
+        :retrun: The list of neighbors
+        :rtype: list :class:`pyrobosim.utils.traversability_graph.Node`
+        """
+        return list(node.neighbors)
+
+    def get_distance(self, nodeA, nodeB):
+        """
+        Returns the distance between nodeA and nodeB
+
+        :param nodeA: The start node
+        :type nodeA: :class: `pyrobosim.utils.traversability_graph.Node`
+        :param nodeB: The end node
+        :type nodeA: :class: `pyrobosim.utils.traversability_graph.Node`
+        :return: The distance between the 2 nodes
+        :rtype: float
+        """
+
+        return nodeA.pose.get_linear_distance(nodeB.pose, ignore_z=True)
+
+    def reset(self):
+        """
+        Resets the graph by clearing out the nodes and edges
+        """
+        self.edges.clear()
+        self.nodes.clear()
```

Comment:
Put periods at the end of all doc sentences, so the main docstring and the `:param:`/`:return:` descriptions (not the types though)

Reply:
> Let's maybe just keep SearchGraph?

Is there any other choices for the name ?
`SearchGraph` does not make much sense, since it is just a graph class and does not have any "search" functionality by itself.
How about just `Graph` ?

Reply:
Do we use this graph for anything other than searching?

I would argue, if anything, that the A* search functionality could be directly added to this graph so you could similarly have something like a `find_path()` method of the graph itself, whch directly calls the python `astar` package. This way, the `WorldGraph` planner doesn't need to have an `AStarGraph` object embedded inside it, but rather can simply use the graph as-is.

Reply:
Another way of thinking about it, right now we have an `AStarGraph` planner that doesn't work standalone because it requires the graph to exist in the first place, and a `WorldGraphPlannerPolygon` planner that wraps around it to make things work.

So my recommendation would be:
* Deprecate `AStarGraph` and just have that planner be focused on occupancy grid only
* Add a `find_path_astar()` method or something to the `SearchGraph` itself so it's actually a "search graph"
* Make `WorldGraphPlannerPolygon` use the above

Reply:
Yea this sounds good, it doesn't look like the AStar graph planner will have any use on its own.
Will move graph astar to the SearchGraph class